R/checks.R

```diff
@@ -531,3 +531,29 @@ format_error_val <- function(x) {
   }
   out
 }
+
+#' @rdname checkmate
+#' @importFrom cli cli_abort
+#' @importFrom rlang caller_arg caller_env try_fetch
+check_arg_is_valid_extension <- function(x,
+                                         valid_extensions,
+                                         arg = caller_arg(x),
+                                         call = caller_env()) {
+  ext <- sub(".*\\.", "", x)
+
+  msg_x <- paste0("Invalid file extension supplied to {.arg {arg}}: ", ext)
+  msg_i <- "File extension must be of type 'xlsx'"
```

Comment:
Nit-picky  but I might go with "File extension must be 'xlsx'" or something like that